packages/docusaurus-plugin-content-docs/src/client/index.ts

```diff
@@ -143,9 +143,16 @@ export function useActiveVersion(
 export function useActiveDocContext(
   pluginId: string | undefined,
 ): ActiveDocContext {
-  const data = useDocsData(pluginId);
-  const {pathname} = useLocation();
-  return getActiveDocContext(data, pathname);
+  try {
+    const data = useDocsData(pluginId);
+    const {pathname} = useLocation();
+    return getActiveDocContext(data, pathname);
+  } catch (error) {
+    throw new Error(
+      'It seems that you are using a doc plugin feature while the doc plugin seems to be disabled in `docusaurus.config.js`.',
```

Comment:
I would write something like this:

```suggestion
      `You are using a feature of the Docusaurus docs plugin, but this plugin does not seem to be enabled${pluginId === Default ? "" : ` (pluginId=${pluginId}`}`,
```

Also I thought maybe this try/catch would be better to add around the `useDocsData` hook instead of `useActiveDocContext`?

Reply:
If other places try to read docs global data, we probably also want to throw them this better error message too so it should be fine 👍